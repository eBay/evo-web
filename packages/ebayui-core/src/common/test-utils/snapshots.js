import { render, prettyDOM } from "@marko/testing-library";
import { expect } from "vitest";
import visualHTML from "visual-html";
import { diffLines } from "diff";

function snapshotHTML() {
    return async (template, input) => {
        const { container } = await render(template, input);
        expect(prettyDOM(container)).toMatchSnapshot();
    };
}

async function diffServerSnapshots(template, input) {
    const { container } = await render(template, input);
    return async (afterTemplate, afterInput) => {
        const { container: afterContainer } = await render(
            afterTemplate,
            afterInput,
        );
        return diffOnlyChanges(container, afterContainer);
    };
}

function diffOnlyChanges(a, b) {
    const changes = diffLines(a, b);
    return changes
        .filter((part) => part.added || part.removed) // omit unchanged lines
        .map((part) => {
            const prefix = part.added ? "+ " : "- ";
            return part.value
                .split("\n")
                .filter((line) => line.trim() !== "") // clean up empty lines
                .map((line) => prefix + line)
                .join("\n");
        })
        .join("\n");
}
function diffHTML(beforeHTML) {
    const before = visualHTML(beforeHTML);
    return (afterHTML) => diffOnlyChanges(before, visualHTML(afterHTML));
}

export {
    snapshotHTML,
    diffServerSnapshots,
    diffHTML,
    diffOnlyChanges,
    visualHTML,
};
