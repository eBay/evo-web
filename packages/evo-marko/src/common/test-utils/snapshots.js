import { render } from "@marko/testing-library";
import { expect } from "vitest";

function snapshotHTML() {
  return async (template, input) => {
    const { container } = await render(template, input);
    expect(container).toMatchSnapshot();
  };
}

export { snapshotHTML };
