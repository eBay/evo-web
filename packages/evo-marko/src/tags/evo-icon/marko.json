{
  "taglib-id": "evo-marko-icons",
  "tags-dir": "./tags"
}
