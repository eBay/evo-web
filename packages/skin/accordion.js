require('./dist/accordion/accordion.css');
