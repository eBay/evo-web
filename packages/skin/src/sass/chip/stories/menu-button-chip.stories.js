export default { title: "Skin/Chip/Menu Button Chip" };

export const collapsed = () => `
<button class="chip chip--filter" aria-expanded="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const expanded = () => `
<button class="chip chip--filter" aria-expanded="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--discovery" aria-expanded="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-chevron-down-12"/>
        </svg>
    </button>
</div>
`;
