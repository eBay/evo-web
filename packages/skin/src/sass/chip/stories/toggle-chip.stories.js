export default { title: "Skin/Chip/Toggle Chip" };

export const button = () => `
<button class="chip" type="button" aria-pressed="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const buttonSelected = () => `
<button class="chip" type="button" aria-pressed="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const anchor = () => `
<a class="chip" herf="https://www.ebay.com">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--selected" herf="https://www.ebay.com">
    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip" type="button" aria-pressed="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-close-12"></use>
        </svg>
    </button>
</div>
`;
