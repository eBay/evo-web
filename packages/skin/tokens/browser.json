{ "dependencies": ["require: ./evo-core.js", "require: ./evo-light.js"] }
