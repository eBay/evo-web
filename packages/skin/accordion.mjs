import './dist/accordion/accordion.css';
