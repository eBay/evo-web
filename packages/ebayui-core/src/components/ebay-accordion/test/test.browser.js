import {
    afterEach,
    beforeEach,
    afterAll,
    beforeAll,
    describe,
    it,
    expect,
} from "vitest";
import { render, fireEvent, cleanup } from "@marko/testing-library";
import { fastAnimations } from "../../../common/test-utils/browser";
import { diffHTML, visualHTML } from "../../../common/test-utils/snapshots";
import { composeStories } from "@storybook/marko";
import * as stories from "../accordion.stories"; // import all stories from the stories file
const { Default, AutoCollapsed } = composeStories(stories);

beforeAll(() => fastAnimations.start());
afterAll(() => fastAnimations.stop());
afterEach(cleanup);
let component;

afterEach(cleanup);

describe("accordion", () => {
    describe("given the accordion in the default state", () => {
        beforeEach(async () => {
            component = await render(Default);
        });

        it("then it matches the snapshot", () => {
            const html = visualHTML(component.container);

            expect(html).toMatchSnapshot();
        });

        describe("when first section toggled", () => {
            let initialHTML;
            beforeEach(async () => {
                initialHTML = diffHTML(component.container);
                await fireEvent.click(component.getByText("Item 1"));
            });

            it("should open the clicked section", async () => {
                expect(initialHTML(component.container)).toMatchSnapshot();
            });

            it("should close an open section when clicked again", async () => {
                await fireEvent.click(component.getByText("Item 1"));
                expect(initialHTML(component.container)).toMatchSnapshot(
                    "Should have no changes from intial",
                );
            });
        });
    });

    describe("given the accordion with autocollapse enabled", () => {
        let initialHTML;
        beforeEach(async () => {
            component = await render(AutoCollapsed);
            initialHTML = diffHTML(component.container);
        });
        describe("when first section toggled", () => {
            beforeEach(async () => {
                await fireEvent.click(component.getByText("Item 1"));
                // Click does not trigger toggle event, need to manually fire it
                await fireEvent(
                    component.getByText("Item 1"),
                    new Event("toggle"),
                );
            });

            it("should open as normal", async () => {
                expect(initialHTML(component.container)).toMatchSnapshot();
            });

            describe("when another section is opened", () => {
                beforeEach(async () => {
                    initialHTML = diffHTML(component.container);
                    // Open second section
                    await fireEvent.click(component.getByText("Item 2"));
                    // Click does not trigger toggle event, need to manually fire it
                    await fireEvent(
                        component.getByText("Item 2"),
                        new Event("toggle"),
                    );
                });
                it("should collapse previous section when new section is opened", async () => {
                    // Verify first section closed and second section opened
                    expect(initialHTML(component.container)).toMatchSnapshot();
                });
            });
        });
    });
});
