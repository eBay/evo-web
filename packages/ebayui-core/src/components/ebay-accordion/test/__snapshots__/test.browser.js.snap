// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`accordion > given the accordion in the default state > then it matches the snapshot 1`] = `
"<div style="
  transition-delay: 0.1s;
  transition-duration: 0.1s
">
  <ul style="
    margin-bottom: 0px;
    margin-top: 0px;
    padding-left: 0px;
    transition-delay: 0.1s;
    transition-duration: 0.1s;
    width: 100%
  ">
    <style scoped>
      ::marker {font-size: 0px}
    </style>
    <li style="
      border-bottom-color: ;
      border-bottom-style: ;
      border-bottom-width: ;
      transition-delay: 0.1s;
      transition-duration: 0.1s
    ">
      <details style="
        transition-delay: 0.1s;
        transition-duration: 0.1s
      ">
        <summary style="
          align-items: center;
          box-sizing: border-box;
          color: var(
                  --details-summary-foreground-color,
                  var(--color-foreground-primary)
              );
          display: flex;
          font-size: var(--font-size-medium);
          font-weight: var(--font-weight-bold);
          justify-content: space-between;
          list-style-position: inside;
          list-style-type: none;
          min-height: 48px;
          padding-bottom: 12px;
          padding-left: 16px;
          padding-right: 16px;
          padding-top: 12px;
          transition-delay: 0.1s;
          transition-duration: 0.1s
        ">
          <style scoped>
            ::before {
              
            }
          </style>
          <span style="
            transition-delay: 0.1s;
            transition-duration: 0.1s
          ">
            Item 1
          </span>
          <span
            hidden
            style="
              display: inline-flex;
              margin-inline-start: 8px;
              transition-delay: 0.1s;
              transition-duration: 0.1s
            "
          >
            <svg
              focusable="false"
              style="
                display: inline-block;
                fill: var(
                        --details-secondary-foreground-color,
                        var(--color-foreground-secondary)
                    );
                height: 16px;
                pointer-events: none;
                stroke-width: 0;
                stroke: currentcolor;
                transition-delay: 0.1s;
                transition-duration: 0.1s;
                vertical-align: middle;
                width: 16px
              "
            >
              <use
                href="#icon-chevron-down-16"
                style="
                  transition-delay: 0.1s;
                  transition-duration: 0.1s
                "
              />
            </svg>
          </span>
        </summary>
        <div style="
          margin-bottom: 6px;
          margin-left: 16px;
          margin-right: 16px;
          opacity: 0;
          transform-origin: center top;
          transform: scaleY(0);
          transition-behavior: ;
          transition-delay: 0.1s;
          transition-duration: 0.1s;
          transition-property: ;
          transition-timing-function: 
        ">
          <p style="
            transition-delay: 0.1s;
            transition-duration: 0.1s
          ">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li style="
      border-bottom-color: ;
      border-bottom-style: ;
      border-bottom-width: ;
      transition-delay: 0.1s;
      transition-duration: 0.1s
    ">
      <details style="
        transition-delay: 0.1s;
        transition-duration: 0.1s
      ">
        <summary style="
          align-items: center;
          box-sizing: border-box;
          color: var(
                  --details-summary-foreground-color,
                  var(--color-foreground-primary)
              );
          display: flex;
          font-size: var(--font-size-medium);
          font-weight: var(--font-weight-bold);
          justify-content: space-between;
          list-style-position: inside;
          list-style-type: none;
          min-height: 48px;
          padding-bottom: 12px;
          padding-left: 16px;
          padding-right: 16px;
          padding-top: 12px;
          transition-delay: 0.1s;
          transition-duration: 0.1s
        ">
          <style scoped>
            ::before {
              
            }
          </style>
          <span style="
            transition-delay: 0.1s;
            transition-duration: 0.1s
          ">
            Item 2
          </span>
          <span
            hidden
            style="
              display: inline-flex;
              margin-inline-start: 8px;
              transition-delay: 0.1s;
              transition-duration: 0.1s
            "
          >
            <svg
              focusable="false"
              style="
                display: inline-block;
                fill: var(
                        --details-secondary-foreground-color,
                        var(--color-foreground-secondary)
                    );
                height: 16px;
                pointer-events: none;
                stroke-width: 0;
                stroke: currentcolor;
                transition-delay: 0.1s;
                transition-duration: 0.1s;
                vertical-align: middle;
                width: 16px
              "
            >
              <use
                href="#icon-chevron-down-16"
                style="
                  transition-delay: 0.1s;
                  transition-duration: 0.1s
                "
              />
            </svg>
          </span>
        </summary>
        <div style="
          margin-bottom: 6px;
          margin-left: 16px;
          margin-right: 16px;
          opacity: 0;
          transform-origin: center top;
          transform: scaleY(0);
          transition-behavior: ;
          transition-delay: 0.1s;
          transition-duration: 0.1s;
          transition-property: ;
          transition-timing-function: 
        ">
          <p style="
            transition-delay: 0.1s;
            transition-duration: 0.1s
          ">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li style="
      transition-delay: 0.1s;
      transition-duration: 0.1s
    ">
      <details style="
        transition-delay: 0.1s;
        transition-duration: 0.1s
      ">
        <summary style="
          align-items: center;
          box-sizing: border-box;
          color: var(
                  --details-summary-foreground-color,
                  var(--color-foreground-primary)
              );
          display: flex;
          font-size: var(--font-size-medium);
          font-weight: var(--font-weight-bold);
          justify-content: space-between;
          list-style-position: inside;
          list-style-type: none;
          min-height: 48px;
          padding-bottom: 12px;
          padding-left: 16px;
          padding-right: 16px;
          padding-top: 12px;
          transition-delay: 0.1s;
          transition-duration: 0.1s
        ">
          <style scoped>
            ::before {
              
            }
          </style>
          <span style="
            transition-delay: 0.1s;
            transition-duration: 0.1s
          ">
            Item 3
          </span>
          <span
            hidden
            style="
              display: inline-flex;
              margin-inline-start: 8px;
              transition-delay: 0.1s;
              transition-duration: 0.1s
            "
          >
            <svg
              focusable="false"
              style="
                display: inline-block;
                fill: var(
                        --details-secondary-foreground-color,
                        var(--color-foreground-secondary)
                    );
                height: 16px;
                pointer-events: none;
                stroke-width: 0;
                stroke: currentcolor;
                transition-delay: 0.1s;
                transition-duration: 0.1s;
                vertical-align: middle;
                width: 16px
              "
            >
              <use
                href="#icon-chevron-down-16"
                style="
                  transition-delay: 0.1s;
                  transition-duration: 0.1s
                "
              />
            </svg>
          </span>
        </summary>
        <div style="
          margin-bottom: 6px;
          margin-left: 16px;
          margin-right: 16px;
          opacity: 0;
          transform-origin: center top;
          transform: scaleY(0);
          transition-behavior: ;
          transition-delay: 0.1s;
          transition-duration: 0.1s;
          transition-property: ;
          transition-timing-function: 
        ">
          <p style="
            transition-delay: 0.1s;
            transition-duration: 0.1s
          ">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
  </ul>
</div>"
`;

exports[`accordion > given the accordion in the default state > when first section toggled > should open the clicked section 1`] = `
"@@ -20,12 +20,16 @@
       transition-delay: 0.1s;
       transition-duration: 0.1s
     ">
-       <details style="
-         transition-delay: 0.1s;
-         transition-duration: 0.1s
-       ">
+       <details
+         open
+         style="
+           transition-delay: 0.1s;
+           transition-duration: 0.1s
+         "
+       >
         <summary style="
           align-items: center;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -61,6 +65,7 @@
             style="
               display: inline-flex;
               margin-inline-start: 8px;
+               transform: rotate(180deg);
               transition-delay: 0.1s;
               transition-duration: 0.1s
             "
@@ -97,9 +102,8 @@
           margin-bottom: 6px;
           margin-left: 16px;
           margin-right: 16px;
-           opacity: 0;
           transform-origin: center top;
-           transform: scaleY(0);
+           transform: scaleY(1);
           transition-behavior: ;
           transition-delay: 0.1s;
           transition-duration: 0.1s;
"
`;

exports[`accordion > given the accordion in the default state > when first section toggled > when another section is opened > should close an open section when clicked again > Should have no changes from initial 1`] = `
"@@ -26,6 +26,7 @@
       ">
         <summary style="
           align-items: center;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
"
`;

exports[`accordion > given the accordion with autocollapse enabled > when first section toggled > should open as normal 1`] = `
"@@ -20,12 +20,16 @@
       transition-delay: 0.1s;
       transition-duration: 0.1s
     ">
-       <details style="
-         transition-delay: 0.1s;
-         transition-duration: 0.1s
-       ">
+       <details
+         open
+         style="
+           transition-delay: 0.1s;
+           transition-duration: 0.1s
+         "
+       >
         <summary style="
           align-items: center;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -61,6 +65,7 @@
             style="
               display: inline-flex;
               margin-inline-start: 8px;
+               transform: rotate(180deg);
               transition-delay: 0.1s;
               transition-duration: 0.1s
             "
@@ -97,9 +102,8 @@
           margin-bottom: 6px;
           margin-left: 16px;
           margin-right: 16px;
-           opacity: 0;
           transform-origin: center top;
-           transform: scaleY(0);
+           transform: scaleY(1);
           transition-behavior: ;
           transition-delay: 0.1s;
           transition-duration: 0.1s;
"
`;

exports[`accordion > given the accordion with autocollapse enabled > when first section toggled > when another section is opened > should collapse previous section when new section is opened 1`] = `
"@@ -20,16 +20,12 @@
       transition-delay: 0.1s;
       transition-duration: 0.1s
     ">
-       <details
-         open
-         style="
-           transition-delay: 0.1s;
-           transition-duration: 0.1s
-         "
-       >
+       <details style="
+         transition-delay: 0.1s;
+         transition-duration: 0.1s
+       ">
         <summary style="
           align-items: center;
-           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -65,7 +61,6 @@
             style="
               display: inline-flex;
               margin-inline-start: 8px;
-               transform: rotate(180deg);
               transition-delay: 0.1s;
               transition-duration: 0.1s
             "
@@ -102,8 +97,9 @@
           margin-bottom: 6px;
           margin-left: 16px;
           margin-right: 16px;
+           opacity: 0;
           transform-origin: center top;
-           transform: scaleY(1);
+           transform: scaleY(0);
           transition-behavior: ;
           transition-delay: 0.1s;
           transition-duration: 0.1s;
@@ -126,12 +122,16 @@
       transition-delay: 0.1s;
       transition-duration: 0.1s
     ">
-       <details style="
-         transition-delay: 0.1s;
-         transition-duration: 0.1s
-       ">
+       <details
+         open
+         style="
+           transition-delay: 0.1s;
+           transition-duration: 0.1s
+         "
+       >
         <summary style="
           align-items: center;
+           background-color: var(--color-state-secondary-hover);
           box-sizing: border-box;
           color: var(
                   --details-summary-foreground-color,
@@ -167,6 +167,7 @@
             style="
               display: inline-flex;
               margin-inline-start: 8px;
+               transform: rotate(180deg);
               transition-delay: 0.1s;
               transition-duration: 0.1s
             "
@@ -203,9 +204,8 @@
           margin-bottom: 6px;
           margin-left: 16px;
           margin-right: 16px;
-           opacity: 0;
           transform-origin: center top;
-           transform: scaleY(0);
+           transform: scaleY(1);
           transition-behavior: ;
           transition-delay: 0.1s;
           transition-duration: 0.1s;
"
`;
